### DISCUSSION TITLE
Highlights Panel

### DISCUSSION THREAD

**Discussion by JodieM** [SFDO-Community/Salesforce-Indicators#74] [Ideas]
There is a Highlights Panel Labs app but it is very simple. Can we make it more complex (read customisable) using the features we have in these apps. 

* Show different values based on different Record Types
* Show different fields based on record Statuses
* Show different labels to the field labels
* Show coloured pills and badges and even icons mixed in with the field values. 
* Display more than 6-10 fields
* Profile (or Custom Permission) based (worst case, Lightning Record Pages allows conditional visibility and you can install multiple Highlights panels)
* Display fields from parent record(s)
* Support text area (and multi-select picklists)
* Conditional visibility of fields (only show when a condition is met or not met - which is what Indicators does)
* Conditional formatting of fields to draw attention to them

Is it just an Indicators panel that is full width with indicators that show text values also. It can use the same CMDT, but the bundle is set to display as a Panel and not the icons. 

NOTE: The labs app just doesn't look right. It needs to look as much as possible like the highlights panel that is there by default (eg shoved hard against the standard collapsed highlights panel).  Also one of the issues with a highlights panel is that the fields look a little floaty as there is no border around each field (but their isn't in a regular LWC using the Lightning Record View Form either)

**Comment by JodieM** [comment]
See also the review of Righteous Lightning Actions https://github.com/SFDO-Community/Salesforce-Indicators/wiki/Righteous-Lightning-Actions and how they do formatted text in there panel. I really want the combination of what they do, and what we can build into a highlights panel with our CMDT.

**Comment by JodieM** [comment]
For advanced options, let's see what we can do to make the panel look and behave like one of those fancy panels they show in every single Salesforce Demo at Dreamforce and other conferences.  
![image](https://github.com/SFDO-Community/Salesforce-Indicators/assets/2966583/6fef0a82-f814-4b20-944a-cc9576a249d9)

**Comment by JodieM** [comment]
Note: We need to put full development of this panel on hold for now, because the Dynamic Forms team have put a Dynamic Highlights Panel on it's prioritization schedule for September 23, so until we know a little more about that, we should wait and see how we can enhance it, or if we just need to rebuild it.

**Comment by tschug** [comment]
Adding these additional items which were discussed in Slack.

- Adding an icon next to fields as well
- Sticky (https://techdicer.com/sticky-header-in-lwc-salesforce/)
- URL support

**Reply by JodieM** [reply]
Oh that is great for our upcoming discussions at Sydney Sprint

**Comment by tschug** [comment]
For reference, we can do a more advanced version than UnofficialSF's

https://unofficialsf.com/highlights-panel-page-component/

**Comment by tschug** [comment]
Someone on Ohana Slack posted another issue:

> There seems to a limitation of real estate from left to right.
If the character count for the field data is longer on one case than another, not all of the fields will show in the Highlights Panel.  And rarely will all of the allowable 10 field show at once, due to the limited space.
>
> > (TIM: I think we know it's limited to 6 anyways)
>
> You can see in the first screenshot, the email address is visible and the account name is slightly shorter.
> In the second screen shot the account name is slightly larger and the email address is dropped from the Highlights Panel.
> The viewing size of the browser window makes no difference on how many of the allowable fields are actually visible in the panel.

Images attached from Ohana Slack

![IMG_3299](https://github.com/SFDO-Community/Salesforce-Indicators/assets/2906839/4311f80b-edb1-42ac-bd06-d97a46740300)

![IMG_3300](https://github.com/SFDO-Community/Salesforce-Indicators/assets/2906839/1e9ccde2-a401-4015-9403-93c269c2a3c6)

**Comment by JodieM** [comment]
This is still a good discussion point. We've added #225 which will go some way to solving this, but there maybe could be an actual highlights panel in future.
